### DISCUSSION TITLE
.rar upload not supported - intentional or bug?

### DISCUSSION THREAD

**Discussion by J3ster1337** [fgilde/MudBlazor.Extensions#39] [General]
Hi again, I'm still researching your library :)

So, when using MudExUploadEdit and trying to upload .rar, this happens: 
![image](https://github.com/fgilde/MudBlazor.Extensions/assets/130294475/fa541f85-5c13-4d1a-a766-5da0d2fc797e)
Note that type is not recognized at all (possibly null or "") and .rar is suggested as usable extension. My current mime types:
```
        "application/zip",
        "application/x-7z-compressed",
        "application/vnd.rar",
        "application/rar",
        "application/x-rar-compressed",
        "application/octet-stream",
        "multipart/x-zip",
        "application/x-zip-compressed"
```
If mime types is not specified, than .rar is not recognized as archive and cannot be previewed:
![image](https://github.com/fgilde/MudBlazor.Extensions/assets/130294475/164d3771-e6b7-4639-aa48-52d6c1a75579)
On preview click nothing happens.

I'm trying to upload file that is generated by WinRar by RMB>"Add To 1.rar". File is not empty.

Is .rar currently not supported or is this a bug?

**Comment by fgilde** [comment]
Upload in general is supported but auto extraction and preview not.  You need to allow "application/x-compressed" or just clear the mimeTypes list and set MimeRestriction to blacklist.

For preview it depends ingeneral on the compression mode that is used. That means currently rar is not supported for FIleDisplay and you should see "The content type application/x-compressed cannot be displayed without a suitable plugin".  

If you have a browser plugin installed that supports x-compressed this will work. 

The current zip preview uses System.Compression.ZipArchive and that can only handle zip files. However I've seen that a nuget package for SevenZip is available that should be able to handle it maybe I can use this instead. 
If you want you can create a Feature request issue and then I will try to implement it sounds funny and makes sense

**Comment by fgilde** [comment]
@J3ster1337 So good news. I've implemented support for rar files as well. Will released in 1.7.67 you can have a preview here [1.7.67-prev-2309011325-main](https://www.nuget.org/packages/MudBlazor.Extensions/1.7.67-prev-2309011325-main) or test it live here https://mudex.azurewebsites.net/upload-edit

**Comment by J3ster1337** [comment]
Thanks! 😄
